### ISSUE TITLE
"Bactopia datasets" showing ERROR

### ISSUE THREAD

**Issue by shankhanath** [bactopia/bactopia#568]
I have successfully installed and created conda environment for bactopia.

But "Bactopia datasets" is giving me error in downloading datasets

```
2024-10-19 21:04:29 INFO     2024-10-19 21:04:29:root:INFO - Downloading amrfinder dataset to /data/sata_data/home/souvik_p1/.bactopia/datasets/v3.1.0/amrfinderplus.tar.gz      datasets.py:162
2024-10-19 21:04:37 INFO     2024-10-19 21:04:37:root:INFO - Downloading mlst dataset to /data/sata_data/home/souvik_p1/.bactopia/datasets/v3.1.0/mlst.tar.gz                    datasets.py:162
2024-10-19 21:04:43 INFO     2024-10-19 21:04:43:root:INFO - Downloading mash dataset to /data/sata_data/home/souvik_p1/.bactopia/datasets/mash-refseq88.k21.msh.xz              datasets.py:162
2024-10-19 21:07:21 INFO     2024-10-19 21:07:21:root:INFO - Downloading sourmash dataset to                                                                                     datasets.py:162
                             /data/sata_data/home/souvik_p1/.bactopia/datasets/gtdb-rs207.genomic-reps.dna.k31.lca.json.gz
2024-10-19 21:07:43 INFO     2024-10-19 21:07:43:root:INFO - Downloading config dataset to null                                                                                  datasets.py:162
                    ERROR    2024-10-19 21:07:43:root:ERROR - Error downloading file, retrying after short sleep.                                                                 datasets.py:87
2024-10-19 21:08:13 ERROR    2024-10-19 21:08:13:root:ERROR - Error downloading file, retrying after short sleep.                                                                 datasets.py:87
2024-10-19 21:09:13 ERROR    2024-10-19 21:09:13:root:ERROR - Error downloading file, retrying after short sleep.                                                                 datasets.py:87
2024-10-19 21:10:43 ERROR    2024-10-19 21:10:43:root:ERROR - Error downloading file, retrying after short sleep.                                                                 datasets.py:87
2024-10-19 21:12:43 ERROR    2024-10-19 21:12:43:root:ERROR - Error downloading file, retrying after short sleep.                                                                 datasets.py:87
2024-10-19 21:15:13 ERROR    2024-10-19 21:15:13:root:ERROR - External command failed with exit code 4!                                                                              utils.py:43

                             Command:
                             bash -c 'wget -O null null'

                             Standard error:
                             --2024-10-19 21:15:13--  http://null/
                             Resolving null (null)... failed: Name or service not known.
                             wget: unable to resolve host address ‘null’

```

when I have rerun the code this is coming
```

2024-10-19 21:29:10 WARNING  2024-10-19 21:29:10:root:WARNING - Found amrfinder at /data/sata_data/home/souvik_p1/.bactopia/datasets/v3.1.0/amrfinderplus.tar.gz, will not       datasets.py:158
                             overwrite unless --force is used.
                    WARNING  2024-10-19 21:29:10:root:WARNING - Found mlst at /data/sata_data/home/souvik_p1/.bactopia/datasets/v3.1.0/mlst.tar.gz, will not overwrite unless    datasets.py:158
                             --force is used.
                    WARNING  2024-10-19 21:29:10:root:WARNING - Found mash at /data/sata_data/home/souvik_p1/.bactopia/datasets/mash-refseq88.k21.msh.xz, will not overwrite     datasets.py:158
                             unless --force is used.
                    WARNING  2024-10-19 21:29:10:root:WARNING - Found sourmash at /data/sata_data/home/souvik_p1/.bactopia/datasets/gtdb-rs207.genomic-reps.dna.k31.lca.json.gz, datasets.py:158
                             will not overwrite unless --force is used.
                    WARNING  2024-10-19 21:29:10:root:WARNING - Found config at null, will not overwrite unless --force is used.                                                 datasets.py:158
```

**Comment by rpetit3** [comment]
Hi @shankhanath 

Thank you for reporting this. 

Can you share the command used? and the type of environment you are running this on? (e.g. desktop/laptop, cluster, etc...) also is an internet connection available?

**Comment by shankhanath** [comment]
I am following your quick start guide.

First I have used the commands 

```
mamba create -y -n bactopia -c conda-forge -c bioconda bactopia
conda activate bactopia
```

Then i have used the command 
`bactopia datasets`

this gave the errors

bactopia environment is successfully created on the conda environment.
I am doing this in cluster and I have internet connection

**Comment by shankhanath** [comment]
Hi @rpetit3
Please let me know if anything else is required.

**Comment by rpetit3** [comment]
Hi @shankhanath 

Let me look into the `bactopia datasets` issue. For the time being you can try `bactopia -profile test,standard`

On you cluster will you use singularity or conda?

**Comment by shankhanath** [comment]
Hello @rpetit3 
Thanks for the reply.. no worries. please take your time to solve this and let me know
I am using conda in my cluster.

I have tried this code that you mentioned.
it gave me this...

```
(bactopia) [souvik_p1@ndmas02 ~]$ bactopia -profile test,standard
2024-10-22 12:44:00 INFO     2024-10-22 12:44:00:root:INFO - Checking if environment pre-builds are needed (this may take a while if building for the first time)                download.py:544
                    INFO     2024-10-22 12:44:00:root:INFO - Begin datasets create to /data/sata_data/home/souvik_p1/.bactopia/conda/bioconda--gnu-wget-1.18                     download.py:261
2024-10-22 12:44:02 ERROR    2024-10-22 12:44:02:root:ERROR - Error creating Conda environment, retrying after short sleep.                                                      download.py:374
2024-10-22 12:44:32 ERROR    2024-10-22 12:44:32:root:ERROR - Error creating Conda environment, retrying after short sleep.                                                      download.py:374
2024-10-22 12:45:32 ERROR    2024-10-22 12:45:32:root:ERROR - Error creating Conda environment, retrying after short sleep.                                                      download.py:374
2024-10-22 12:47:02 ERROR    2024-10-22 12:47:02:root:ERROR - Error creating Conda environment, retrying after short sleep.                                                      download.py:374
2024-10-22 12:49:02 ERROR    2024-10-22 12:49:02:root:ERROR - Error creating Conda environment, retrying after short sleep.                                                      download.py:374
2024-10-22 12:51:32 ERROR    2024-10-22 12:51:32:root:ERROR - External command failed with exit code 109!                                                                            utils.py:43

                             Command:
                             bash -c 'mamba create -y -p /data/sata_data/home/souvik_p1/.bactopia/conda/bioconda--gnu-wget-1.18 -c conda-forge -c bioconda --force
                             bioconda::gnu-wget=1.18'

                             Standard error:
                             The following argument was not expected: --force
                             Run with --help for more information.

 N E X T F L O W   ~  version 24.04.4

Launching `/data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0/main.nf` [jolly_visvesvaraya] DSL2 - revision: 0cd9f79ba7

MODULES: [datasets, gather, qc, assembler, sketcher, amrfinderplus, mlst]
MODULES: [datasets, gather, qc, assembler, sketcher, amrfinderplus, mlst]


---------------------------------------------
   _                _              _
  | |__   __ _  ___| |_ ___  _ __ (_) __ _
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` |
  | |_) | (_| | (__| || (_) | |_) | | (_| |
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|
                            |_|
  bactopia v3.1.0
  Bactopia is a flexible pipeline for complete analysis of bacterial genomes
---------------------------------------------
Core Nextflow options
  runName              : jolly_visvesvaraya
  container            : quay.io/bactopia/bactopia:3.1.0
  launchDir            : /data/sata_data/home/souvik_p1
  workDir              : /data/sata_data/home/souvik_p1/work
  projectDir           : /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0
  userName             : souvik_p1
  profile              : test,standard
  configFiles          : /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0/nextflow.config

Required Parameters
  r1                   : https://github.com/bactopia/bactopia-tests/raw/main/data/species/portiera/illumina/SRR2838702_R1.fastq.gz
  r2                   : https://github.com/bactopia/bactopia-tests/raw/main/data/species/portiera/illumina/SRR2838702_R2.fastq.gz
  sample               : SRR2838702

Dataset Parameters
  genome_size          : 358242

QC Parameters
  adapters             : /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0/data/EMPTY_ADAPTERS
  phix                 : /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0/data/EMPTY_PHIX

Prokka Parameters
  proteins             : /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/share/bactopia-3.1.0/data/proteins.faa

Max Job Request Parameters
  max_cpus             : 2
  max_memory           : 6 GB
  max_time             : 2h

Nextflow Profile Parameters
  condadir             : /data/sata_data/home/souvik_p1/.bactopia/conda
  datasets_cache       : /data/sata_data/home/souvik_p1/.bactopia/datasets
  singularity_cache_dir: /data/sata_data/home/souvik_p1/.bactopia/singularity

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
WARN:
WARN: Conda Disclaimer
WARN:
WARN: If you have access to Docker or Singularity, please consider
WARN: running Bactopia using containers. The containers are less
WARN: susceptible to Conda environment related issues (e.g. version
WARN: conflicts) and errors caused by creation of conda environments
WARN: in parallel (use '--max_cpus 1' to over come this error).
WARN:
WARN: To use containers, you can use the profile parameter
WARN:     Docker: -profile docker
WARN:     Singularity: -profile singularity
WARN:
--------------------------------------------------------------------
[skipped  ] process > BACTOPIA:DATASETS                        [100%] 1 of 1, stored: 1 ✔
[-        ] process > BACTOPIA:GATHER:GATHER_MODULE            -
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT             -
[-        ] process > BACTOPIA:QC:QC_MODULE                    -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE      -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT          -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE        -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE         -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN -
[-        ] process > BACTOPIA:AMRFINDERPLUS:CSVTK_CONCAT      -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS            -
Creating env using conda: bioconda::bactopia-gather=1.0.4 [cache /data/sata_data/home/souvik_p1/.bactopia/conda/env-a738b058e7464f0a589401ac609ecaf9]
Exception in thread "Thread-2" groovy.lang.GroovyRuntimeException: exception while reading process stream
[skipped  ] process > BACTOPIA:DATASETS                        [100%] 1 of 1, stored: 1 ✔
[-        ] process > BACTOPIA:GATHER:GATHER_MODULE            -
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT             -
[-        ] process > BACTOPIA:QC:QC_MODULE                    -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE      -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT          -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE        -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE         -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN -
[-        ] process > BACTOPIA:AMRFINDERPLUS:CSVTK_CONCAT      -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS            -

ERROR ~ Error executing process > 'BACTOPIA:GATHER:GATHER_MODULE (1)'

Caused by:
  Failed to create Conda environment
    command: conda create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /data/sata_data/home/souvik_p1/.bactopia/conda/env-a738b058e7464f0a589401ac609ecaf9 bioconda::bactopia-gather=1.0.4
    status : 143
    message:
      Retrieving notices: ...working... /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'repo.anaconda.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'repo.anaconda.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      done
      Collecting package metadata (current_repodata.json): ...working... done
      Solving environment: ...working... unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
      Collecting package metadata (repodata.json): ...working... done
      Solving environment: ...working...



 -- Check '.nextflow.log' file for details

    Bactopia Execution Summary
    ---------------------------
[skipped  ] process > BACTOPIA:DATASETS                        [100%] 1 of 1, stored: 1 ✔
[-        ] process > BACTOPIA:GATHER:GATHER_MODULE            -
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT             -
[-        ] process > BACTOPIA:QC:QC_MODULE                    -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE      -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT          -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE        -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE         -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN -
[-        ] process > BACTOPIA:AMRFINDERPLUS:CSVTK_CONCAT      -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS            -
ERROR ~ Error executing process > 'BACTOPIA:GATHER:GATHER_MODULE (1)'

Caused by:
  Failed to create Conda environment
    command: conda create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /data/sata_data/home/souvik_p1/.bactopia/conda/env-a738b058e7464f0a589401ac609ecaf9 bioconda::bactopia-gather=1.0.4
    status : 143
    message:
      Retrieving notices: ...working... /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'repo.anaconda.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'repo.anaconda.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      /data/sata_data/home/souvik_p1/miniconda3/envs/bactopia/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'conda.anaconda.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
        warnings.warn(
      done
      Collecting package metadata (current_repodata.json): ...working... done
      Solving environment: ...working... unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
      Collecting package metadata (repodata.json): ...working... done
      Solving environment: ...working...



 -- Check '.nextflow.log' file for details

```

**Comment by rpetit3** [comment]
Can you try my dev build? I made some updates for the conda side of things:

```
mamba create -n bactopia -c conda-forge -c bioconda bactopia 'bactopia-py>=1.3.0'
```

Otherwise, do you have access to Singularity on your cluster? Also, what's the scheduler (SLURM, LSF, etc...)?

**Comment by shankhanath** [comment]
Hi..
I do not have access to singularity now. but talked with our IT admin they said they can help me for this.
However I have permission for docker. will that be a help ?

One thing I have told you wrong. We have standalone servers. not cluster. so there is no scheduler in this.
I am actually working on server.

**Comment by rpetit3** [comment]
Oh, if you have Docker, just add `-profile docker` and you will be all set.

In the above command, `--R1` and `--R2` should be `--r1` and `--r2`. The input FASTQs will also need to be compressed. Also, `--datasets` is no longer needed.


```
bactopia \
  --r1 SRR23106201_1.fastq.gz \
  --r2 SRR23106201_2.fastq.gz \
  --sample SRR23106201  \
  --outdir SRR23106201_output \
 -profile docker
```

**Comment by shankhanath** [comment]
Hi @rpetit3 all is working fine.
Just need another small help.
I have some fasta files also. from ENA database. Can those fasta files be analysed using this code using docker? can you provide some code like this. please let me know

**Comment by rpetit3** [comment]
Hi @shankhanath 

Sorry for the slow reply!

For local assemblies (fasta files), check out https://bactopia.github.io/latest/beginners-guide/#assembly

If remote assemblies and you know the NCBI Assembly accession, see https://bactopia.github.io/latest/beginners-guide/#ncbi-assembly-accession

**Comment by rpetit3** [comment]
Hi everyone, 

I've released [v3.2.0](https://github.com/bactopia/bactopia/releases/tag/v3.2.0), which includes a fix for this issue.  If the issue persists, please let me know. 

Thank you for reporting this,
Robert

**Comment by shankhanath** [comment]
Hi @rpetit3 
hope you are doing well..
So what I have to do for this? should I reinstall Bactopia for this? and then run using docker.. 
i have now downloaded the source code zip file and extracted..

**Comment by rpetit3** [comment]
Hi @shankhanath 

I recommend reinstalling bactopia to make sure you get v3.2.0, then yes, you can use docker.

Cheers,
Robert

**Comment by shankhanath** [comment]
Hi @rpetit3 Thanks a ton for this..

I have installed it and ran the code.. but dont know why all the modules are not running.. can you see and help it..  i am bothering you so much..  

```
(base) [wsjuly24@ndmas02 ODOG]$ nextflow run /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf   --r1 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R1_001.fastq.gz   --r2 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R2_001.fastq.gz   --sample CI1-PS186CI1   --outdir /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new   -profile docker
Nextflow 24.10.5 is available - Please consider updating your version to it

 N E X T F L O W   ~  version 24.04.4

Launching `/data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf` [grave_kirch] DSL2 - revision: 0cd9f79ba7



---------------------------------------------
   _                _              _
  | |__   __ _  ___| |_ ___  _ __ (_) __ _
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` |
  | |_) | (_| | (__| || (_) | |_) | | (_| |
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|
                            |_|
  bactopia v3.2.0
  Bactopia is a flexible pipeline for complete analysis of bacterial genomes
---------------------------------------------
Core Nextflow options
  runName          : grave_kirch
  containerEngine  : docker
  container        : quay.io/bactopia/bactopia:3.2.0
  launchDir        : /data/sata_data/home/wsjuly24/ODOG
  workDir          : /data/sata_data/home/wsjuly24/ODOG/work
  projectDir       : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0
  userName         : wsjuly24
  profile          : docker
  configFiles      : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/nextflow.config

Required Parameters
  r1               : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R1_001.fastq.gz
  r2               : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R2_001.fastq.gz
  sample           : CI1-PS186CI1

Dataset Parameters
  genome_size      : 0

QC Parameters
  adapters         : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/EMPTY_ADAPTERS
  phix             : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/EMPTY_PHIX

Prokka Parameters
  proteins         : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/proteins.faa

Optional Parameters
  outdir           : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new

Max Job Request Parameters
  max_memory       : 64 GB
  max_time         : 2h

Nextflow Profile Parameters
  condadir         : /data/sata_data/home/wsjuly24/.bactopia/conda
  datasets_cache   : /data/sata_data/home/wsjuly24/.bactopia/datasets
  singularity_cache: /data/sata_data/home/wsjuly24/.bactopia/singularity

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
executor >  local (6)
[8e/271f5c] process > BACTOPIA:DATASETS                                  [100%] 1 of 1 ✔
executor >  local (6)
[8e/271f5c] process > BACTOPIA:DATASETS                                  [100%] 1 of 1 ✔
[d6/d66782] process > BACTOPIA:GATHER:GATHER_MODULE (CI1-PS186CI1)       [100%] 1 of 1 ✔
[d3/346901] process > BACTOPIA:GATHER:CSVTK_CONCAT (meta)                [100%] 1 of 1 ✔
[01/09c6a4] process > BACTOPIA:QC:QC_MODULE (CI1-PS186CI1)               [100%] 1 of 1 ✔
[94/a908b8] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE (CI1-PS186CI1) [100%] 1 of 1 ✔
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT                    -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE                  -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE                   -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN           -
[-        ] process > BACTOPIA:AMRFINDERPLUS:CSVTK_CONCAT                -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                          -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT                         -
[62/4eb899] process > BACTOPIA:DUMPSOFTWAREVERSIONS (1)                  [100%] 1 of 1 ✔

    Bactopia Execution Summary
    ------------------------------
    Workflow         : bactopia
    Bactopia Version : 3.2.0
    Nextflow Version : 24.04.4
    Command Line     : nextflow run /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf --r1 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R1_001.fastq.gz --r2 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186/CI1-PS186_S298_L002_R2_001.fastq.gz --sample CI1-PS186CI1 --outdir /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new -profile docker
    Profile          : docker
    Resumed          : false
    Completed At     : 2025-03-25T12:59:37.614882+05:30
    Duration         : 28m 31s
    Success          : true
    Exit Code        : 0
    Error Report     : -
    Launch Dir       : /data/sata_data/home/wsjuly24/ODOG
    Merged Results   : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new/bactopia-runs/bactopia-20250325-123105

    Further analyze your samples using Bactopia Tools, with the following command:
    --------------------------------------------------------------------------------
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new --wf <REPLACE_WITH_BACTOPIA_TOOL_NAME>

    Examples:
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new --wf pangenome
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new --wf merlin
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI1-PS186CI1_output_new --wf sccmec

    See the full list of available Bactopia Tools: bactopia --list_wfs

Completed at: 25-Mar-2025 12:59:38
Duration    : 28m 32s
CPU hours   : 1.3
Succeeded   : 6
```

**Comment by rpetit3** [comment]
Hi @shankhanath,

Please check the assembler folder for that sample. It looks like the assembly might not have been good enough for down stream analysis.

Cheers,
Robert

**Comment by shankhanath** [comment]
Hi @rpetit3 

I have used another sample and again encountered same issue.. 

```
(base) [wsjuly24@ndmas02 ODOG]$ nextflow run /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf \
>   --r1 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R1_001.fastq.gz \
>   --r2 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R2_001.fastq.gz \
>   --sample CI7-PS186CI1 \
>   --outdir /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output \
>   -profile docker
Nextflow 24.10.5 is available - Please consider updating your version to it

 N E X T F L O W   ~  version 24.04.4

Launching `/data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf` [naughty_fourier] DSL2 - revision: 0cd9f79ba7



---------------------------------------------
   _                _              _
  | |__   __ _  ___| |_ ___  _ __ (_) __ _
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` |
  | |_) | (_| | (__| || (_) | |_) | | (_| |
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_|
                            |_|
  bactopia v3.2.0
  Bactopia is a flexible pipeline for complete analysis of bacterial genomes
---------------------------------------------
Core Nextflow options
  runName          : naughty_fourier
  containerEngine  : docker
  container        : quay.io/bactopia/bactopia:3.2.0
  launchDir        : /data/sata_data/home/wsjuly24/ODOG
  workDir          : /data/sata_data/home/wsjuly24/ODOG/work
  projectDir       : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0
  userName         : wsjuly24
  profile          : docker
  configFiles      : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/nextflow.config

Required Parameters
  r1               : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R1_001.fastq.gz
  r2               : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R1_001.fastq.gz
  sample           : CI7-PS186CI1

Dataset Parameters
  genome_size      : 0

QC Parameters
  adapters         : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/EMPTY_ADAPTERS
  phix             : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/EMPTY_PHIX

Prokka Parameters
  proteins         : /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/data/proteins.faa

Optional Parameters
  outdir           : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output

Max Job Request Parameters
  max_memory       : 64 GB
  max_time         : 2h

Nextflow Profile Parameters
  condadir         : /data/sata_data/home/wsjuly24/.bactopia/conda
  datasets_cache   : /data/sata_data/home/wsjuly24/.bactopia/datasets
  singularity_cache: /data/sata_data/home/wsjuly24/.bactopia/singularity

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
executor >  local (5)
[skipped  ] process > BACTOPIA:DATASETS                                  [100%] 1 of 1, stored: 1 ✔
executor >  local (5)
[skipped  ] process > BACTOPIA:DATASETS                                  [100%] 1 of 1, stored: 1 ✔
[a9/68e79b] process > BACTOPIA:GATHER:GATHER_MODULE (CI7-PS186CI1)       [100%] 1 of 1 ✔
[99/394a5c] process > BACTOPIA:GATHER:CSVTK_CONCAT (meta)                [100%] 1 of 1 ✔
[80/4686e7] process > BACTOPIA:QC:QC_MODULE (CI7-PS186CI1)               [100%] 1 of 1 ✔
[6b/c856c3] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE (CI7-PS186CI1) [100%] 1 of 1 ✔
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT                    -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE                  -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE                   -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN           -
[-        ] process > BACTOPIA:AMRFINDERPLUS:CSVTK_CONCAT                -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                          -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT                         -
[e3/c0d221] process > BACTOPIA:DUMPSOFTWAREVERSIONS (1)                  [100%] 1 of 1 ✔

    Bactopia Execution Summary
    ------------------------------
    Workflow         : bactopia
    Bactopia Version : 3.2.0
    Nextflow Version : 24.04.4
    Command Line     : nextflow run /data/sata_data/home/wsjuly24/bactopia/bactopia-3.2.0/main.nf --r1 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R1_001.fastq.gz --r2 /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186/CI7-PS186_S446_L003_R1_001.fastq.gz --sample CI7-PS186CI1 --outdir /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output -profile docker
    Profile          : docker
    Resumed          : false
    Completed At     : 2025-03-25T13:24:19.952011+05:30
    Duration         : 19m 7s
    Success          : true
    Exit Code        : 0
    Error Report     : -
    Launch Dir       : /data/sata_data/home/wsjuly24/ODOG
    Merged Results   : /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output/bactopia-runs/bactopia-20250325-130511

    Further analyze your samples using Bactopia Tools, with the following command:
    --------------------------------------------------------------------------------
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output --wf <REPLACE_WITH_BACTOPIA_TOOL_NAME>

    Examples:
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output --wf pangenome
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output --wf merlin
    bactopia -profile docker --bactopia /data/sata_data/home/wsjuly24/ODOG/ODOG/CI7-PS186CI1_output --wf sccmec

    See the full list of available Bactopia Tools: bactopia --list_wfs

Completed at: 25-Mar-2025 13:24:20
Duration    : 19m 7s
CPU hours   : 1.3
Succeeded   : 5
```

This time I have gone for an online tool like BVBRC which gave me very good taxonomy (_Rothia amarae_) as well as genome assembly is also successfully done by that tool.. 
I dont know what is happening.. I have checked the assembly results and saw the error text..
This is the error
> CI7-PS186CI1 assembled successfully, but 0 contigs were formed. Please investigate
> CI7-PS186CI1 to determine a cause (e.g. metagenomic, contaminants, etc...) for this
> outcome. Further assembly-based analysis of CI7-PS186CI1 will be discontinued.

But interestingly in BVBRC it gave 18 contigs and all the relevant genome assembly results with 100% completness..... I have added the genome quality report from BVBRC...

[genome_quality_details.txt](https://github.com/user-attachments/files/19465943/genome_quality_details.txt)

**Comment by rpetit3** [comment]
Hi @shankhanath 

Can you share details about your sample? (species, sequencing technology, coverage, etc...) Also can you pull out [logs](https://bactopia.github.io/latest/bactopia/assembler/#audit-trail) for the assembly step and share them?

Thank you!
Robert

**Comment by shankhanath** [comment]
Hi @rpetit3 

I have done WGS in NovaSeq6000 using 2x150 bp chemistry.. I am sharing the file details which is publicly available.. you can also download the data from there and check..

https://ibdc.dbtindia.gov.in/inda/view_sample?sampleid=INS0016143


I have also added the log file generated in the assembler folder
[nf-assembler.log](https://github.com/user-attachments/files/19469534/nf-assembler.log)
[shovill.log](https://github.com/user-attachments/files/19469535/shovill.log)

**Comment by rpetit3** [comment]
Hi @shankhanath 

In the Shovill log I see:

```
[skesa] Memory provided is insufficient to do runs in 10 cycles for the read coverage. We find that 16 Gb for 20x coverage of a 5 Mb genome is usually sufficient
```
